src/platform/win32localfilesystem.cpp

```diff
@@ -29,11 +30,19 @@
 
 File *Win32LocalFileSystem::Open_File(const char *filename, int mode)
 {
-    if (strlen(filename) <= 0) {
+    using Win32BufferedFile = Thyme::Win32BufferedFile;
+
+    if (filename == nullptr || *filename == '\0') {
         return nullptr;
     }
 
-    Win32LocalFile *file = NEW_POOL_OBJ(Win32LocalFile);
+    File *file;
+
+    if (mode & File::BUFFERED) {
+        file = NEW_POOL_OBJ(Win32BufferedFile);
+    } else {
+        file = NEW_POOL_OBJ(Win32LocalFile);
+    }
```

Comment:
Only uses `Win32BufferedFile` when the flag is set, otherwise uses legacy `Win32LocalFile`.